doc: guides: flash_debug: Improve readability and add debug probe and host tools compatibility table

Julia-Hathaway opened the pull request:
In host tools, reformats the SAM-BA flash configuration section to use tabs for
different devices.
In debug probes, adds a compatibility table between debug probes and host tools 
and reformats ST-Link flash and debug section with tabs for each host tool option.

Signed-off-by: Julia Hathaway <julia.hathaway@nxp.com>

Julia-Hathaway committed:
doc/host-tools: Add tabs to SAM-BA section to improve readability

Reformats the three sections on flash configuration for devices with
different bootloader compatibility into tabs to streamline instructions

Signed-off-by: Julia Hathaway <julia.hathaway@nxp.com>

Julia-Hathaway committed:
doc/probes: Add debug probe and host tools table and ST-Link tabs

Includes a table to display compatibility between debug probes and host
tools.
Moves the J-Link External Debug Probe section up for consistency with
host tools listings and to clarify separation between sections.
Reformats Flash and debug with ST-Link section with tabs for different
host tool options for clarity and readability.

Signed-off-by: Julia Hathaway <julia.hathaway@nxp.com>

nandojve reviewed (approved):
Hi Julia ,

Very nice visual improvement.

Thank you!

mbolivar-nordic reviewed (approved):
> Very nice visual improvement.

Agreed!

Julia-Hathaway committed:
samples/hci_uart: doc: Fix reference to debug probes page

Changes reference link to the ST-Link debug probes section to match
new labeling.

Signed-off-by: Julia Hathaway <julia.hathaway@nxp.com>
